### ISSUE TITLE
Allow use as a library?

### ISSUE THREAD

**Issue by jelder** [sbdchd/squawk#333]
Squawk is currently only usable as a CLI tool. What is the level of interest in making it available as a library as well?

**Comment by chdsbd** [comment]
I think we used to publish crates to crates.io, but got rid of that because it was a little complex and unnecessary for the CLI tool.

I'm not opposed to publishing a library again though.

**Comment by psteinroe** [comment]
at some point i would like to integrate this into the Postgres language server I am building. A library would make this significantly easier.

**Comment by sbdchd** [comment]
I've released Squawk 2.0.0 which has a new, error tolerant parser that makes it a good fit for a language server.

I think we could publish the `crates/squawk_syntax` to crates.io.

Currently the `crates/squawk_linter` is layered on top of `crates/squawk_syntax`.

`crates/squawk_syntax` is itself a layer (an AST) on top of CST provided by `crates/squawk_parser`, which actually contains the bulk of logic

**Comment by psteinroe** [comment]
oh wow, that's impressive! thanks for sharing. Will definitely check it out. 

Did you write that 14k lines by hand, or did you generate it from the syntax in the docs?

**Comment by sbdchd** [comment]
Yeah wrote it by hand using the syntax from the docs & looking at the [postgres gram.y](https://github.com/postgres/postgres/blob/69aca072ebf64dcf36958b6958ce4388480fcba2/src/backend/parser/gram.y)
The docs are missing stuff so the grammar is usually better to look at.

Been working on it off and on since the Fall!

**Comment by lubo** [comment]
I'd also like to see this published on crates.io, so that I can just run `cargo install` on it.

**Comment by lubo** [comment]
The CLI.

**Comment by lubo** [comment]
What's the status on this? I've just added `squawk` crate to my project from Git and I'd like to be able to leverage Dependabot for version upgrades, which it can only do with registry sources. I've seen 2c2660f959af31dc9b9f732f4483a45a30c58940. @sbdchd, have you head chance to think about it? What's actually the problem? Is there something I can do to make it happen?

**Comment by sbdchd** [comment]
When I tried to set it up last time it ended up being more work than I anticipated, but I'm cool with publishing the crates to crates.io!

**Comment by lubo** [comment]
Have you tried using `--dry-run` with the publish command? Looks like it isn't used in the CI pipeline for non-release runs and it seems like you've been having a lot of issues with this that could be easily uncovered with dry run.
